xgboost_ray/main.py

```diff
@@ -61,11 +68,15 @@ def _start_rabit_tracker(num_workers: int):
     rabit_tracker.start(num_workers)
 
     # Wait until context completion
-    thread = Thread(target=rabit_tracker.join)
-    thread.daemon = True
-    thread.start()
+    process = multiprocessing.Process(target=rabit_tracker.join)
```

Comment:
So the Rabit tracker is the "driver"/head process of the Rabit allreduce implementation. All Rabit actors register with the tracker.
We're running the tracker on the driver and not in a Ray process or actor. After pausing or killing actors it can lead to problems if we re-use the same tracker instance, so we have to create a new one. In order to properly clean up the old tracker, we have to terminate the process somehow. Since python threads don't support killing, we're starting a process here that is hence killable.

Reply:
Got it that makes sense! Do you think you can add this as a comment in the code?

Reply:
yep, had the same question too! A comment in the code would be much appreciated.